analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -321,6 +321,35 @@ class Arithmetic
     int? MultiplicationAndAssignments2(int? i) => (i = null) * (i = 42) * i.Value; // Compliant
 }
 
+class Comparisons
+{
+    void NullIsNeitherSmallerNorBiggerThanANumber(int? i)
+    {
+        i = null;
+        if (i > 0) _ = i.Value;           // Noncompliant, FP: positive, therefore non-empty
+
+        i = null;
+        if (i < 0) _ = i.Value;           // Noncompliant, FP: negative, therefore non-empty
+
+        i = null;
+        if (i == 42) _ = i.Value;         // Compliant, 42, therefore non-empty
+
+        i = Unknown();
+        if (i >= 0 || i < 0)
+            _ = i.Value;                  // Compliant, non-empty
+        else
+            _ = i.Value;                  // FN, empty
```

Comment:
This is not a nullable test, but a test of SMT capabilities.

To test `||` with nullable, there should be something like
```
if(i==0 || i==-1)
  // FP here
```

Reply:
`if (i == 0 || i == 1) _ = i.Value;` added to the tests.

Concerning `i >= 0 || i < 0`: can you please elaborate on the why this is not a nullable test? The method where the test was added is called `NullIsNeitherSmallerNorBiggerThanANumber`: it tests that `null` is not in order relation with any number, which implies that `i >= 0 || i < 0` is semantically equivalent to `i.HasValue`, and should be treated the same.

We have tests for `i.HasValue` in this test suite, since it constrains non-nullability on `i`. 
A condition such as `i >= 0 || i < 0` should add a non-nullable constraint on `i`, the same `i.HasValue` does.

Reply:
Because it tests the satisfiability of the expression `i==0 || i<0` which is equivalent to `i is int`.

This will fail because we can't detect the satisfiability until we replace CBDE. Not because we need changes related to the nullability.

Reply:
Ok, I have removed it.

However, if not here, something, somewhere else, should test this category of cases, even while CBDE is in force.

Regardless, I would consider proper null-inference for `i >=0 || i < 0` relevant for nullable value types because comparisons operators such as `>=` and `<` have special built-in operators, which lift the comparison inside the `Nullable` wrapper, and that specifically happens for `Nullable<T>`, and for no other type with built-in comparison operators.